src/table/resizer/index.tsx

```diff
@@ -97,83 +108,138 @@ export function Resizer({
+          setIsKeyboardDragging(false);
+          onFinishStable();
+        }
+        if (event.keyCode === KeyCode.escape) {
+          setIsKeyboardDragging(false);
+          resetColumnWidth();
+        }
+      } else {
+        // Enter keyboard dragging mode
+        if (event.keyCode === KeyCode.enter || event.keyCode === KeyCode.space) {
+          event.preventDefault();
+          setIsKeyboardDragging(true);
+        }
       }
     };
 
-    updateTrackerPosition(headerCell.getBoundingClientRect().right);
-    document.body.classList.add(styles['resize-active']);
-    document.body.classList.remove(styles['resize-active-with-focus']);
+    return { updateTrackerPosition, updateColumnWidth, resetColumnWidth, onMouseMove, onMouseUp, onKeyDown };
+  }, [headerCell, isKeyboardDragging, minWidth, onDragStable, onFinishStable]);
+
+  useEffect(() => {
+    if ((!isDragging && !resizerHasFocus) || !headerCell || !handlers) {
+      return;
+    }
+
+    originalHeaderCellWidthRef.current = headerCell.getBoundingClientRect().width;
+
+    handlers.updateTrackerPosition(headerCell.getBoundingClientRect().right);
+
     if (isDragging) {
-      document.addEventListener('mousemove', onMouseMove);
-      document.addEventListener('mouseup', onMouseUp);
+      document.body.classList.add(styles['resize-active']);
+      document.addEventListener('mousemove', handlers.onMouseMove);
+      document.addEventListener('mouseup', handlers.onMouseUp);
     }
     if (resizerHasFocus) {
       document.body.classList.add(styles['resize-active-with-focus']);
-      headerCell.addEventListener('keydown', onKeyDown);
+      headerCell.addEventListener('keydown', handlers.onKeyDown);
+    }
+    if (isKeyboardDragging) {
+      document.body.classList.add(styles['resize-active']);
     }
 
     return () => {
       clearTimeout(autoGrowTimeout.current);
       document.body.classList.remove(styles['resize-active']);
-      document.removeEventListener('mousemove', onMouseMove);
-      document.removeEventListener('mouseup', onMouseUp);
-      headerCell.removeEventListener('keydown', onKeyDown);
+      document.body.classList.remove(styles['resize-active-with-focus']);
+      document.removeEventListener('mousemove', handlers.onMouseMove);
+      document.removeEventListener('mouseup', handlers.onMouseUp);
+      headerCell.removeEventListener('keydown', handlers.onKeyDown);
     };
-  }, [headerCell, isDragging, onDragStable, onFinishStable, resizerHasFocus, minWidth]);
-  return (
-    <span
-      className={clsx(
-        styles.resizer,
-        isDragging && styles['resizer-active'],
-        (resizerHasFocus || showFocusRing) && styles['has-focus']
-      )}
-      onMouseDown={event => {
-        if (event.button !== 0) {
-          return;
+  }, [headerCell, isDragging, isKeyboardDragging, onFinishStable, resizerHasFocus, handlers]);
+
+  const resizerWidthId = useUniqueId();
+  const resizerRole = isKeyboardDragging ? 'separator' : 'button';
```

Comment:
I have some concerns to change the role with state changing. Will it be confusing by users? 
In addition, the button is labelled by header name without any indication it is used to resize the cell.

Reply:
I think that should not matter much whether it is the same element or, say, a focus transition from one element (button) to another (separator). The button is labelled the same way as the separator was. Probably a specific label or, say, ARIA role-description would make sense here, but am not sure if that is really necessary as so far we don't have it.

Reply:
Yes, I am planning to go this way instead of adding the description preemptively.